### PR TITLE
Optional parameters are implicitly nullable

### PR DESCRIPTION
**Pull request by bgribaudo**
Revising to reflect that a function type parameter which is defined as optional is also automatically made nullable.

```powerquery-m
let
    Type = type function (optional x as text) as any,
    ParameterType = Type.FunctionParameters(Type)[x]
in
    Type.IsNullable(ParameterType) // returns true
```

### CODE REVIEW

**Comment by JasonWHowell** [comment]
@ehrenMSFT could you please review this improvement to the doc about Types? Thanks~ Jason

File: query-languages/m/m-spec-types.md
Comment (ehrenMSFT) [inline query-languages/m/m-spec-types.md:244]: It seems like this should go at the end of the section, rather than here.

Comment (ehrenMSFT) [inline query-languages/m/m-spec-types.md:244]:
```suggestion
Specifying a parameter as optional implicitly makes its type nullable. The following create identical function types:
```

**Review by ehrenMSFT — CHANGES_REQUESTED** [review]

File: query-languages/m/m-spec-types.md
Comment (bgribaudo) [reply query-languages/m/m-spec-types.md]: Thanks, @ehrenMSFT! Moved. How does it look now?

**Comment by ehrenMSFT** [comment]
#sign-off

**Comment by Jak-MS** [comment]
@JasonWHowell @ehrenMSFT - the PR review team is waiting on perms to be granted before we can merge this PR. 

fyi @DougKlopfenstein -

**Comment by Jak-MS** [comment]
#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

**Pull request merged**
